contracts/MechMarketplace.sol

```diff
@@ -548,69 +522,41 @@ contract MechMarketplace is IErrorsMarketplace {
         ));
     }
 
-    /// @dev Checks for service validity and optionally for service staking correctness.
-    /// @param stakingInstance Staking instance address.
+    /// @dev Checks for service validity.
     /// @param serviceId Service Id.
     /// @return multisig Service multisig address.
     function checkServiceAndGetMultisig(
-        address stakingInstance,
         uint256 serviceId
     ) public view returns (address multisig) {
         // Check mech service Id
-        if (stakingInstance == address(0)) {
-            IServiceRegistry.ServiceState state;
-            (, multisig, , , , , state) = IServiceRegistry(serviceRegistry).mapServices(serviceId);
-            if (state != IServiceRegistry.ServiceState.Deployed) {
-                revert WrongServiceState(uint256(state), serviceId);
-            }
-        } else {
-            // Check staking instance
-            checkStakingInstance(stakingInstance, serviceId);
-
-            // Get the staked service info for the mech
-            IStaking.ServiceInfo memory serviceInfo = IStaking(stakingInstance).getServiceInfo(serviceId);
-            multisig = serviceInfo.multisig;
-        }
-    }
-
-    /// @dev Checks for staking instance contract validity.
-    /// @param stakingInstance Staking instance address.
-    /// @param serviceId Service Id.
-    function checkStakingInstance(address stakingInstance, uint256 serviceId) public view {
-        // Check that the mech staking instance is valid
-        if (!IStakingFactory(stakingFactory).verifyInstance(stakingInstance)) {
-            revert UnauthorizedAccount(stakingInstance);
-        }
-
-        // Check if the mech service is staked
-        IStaking.StakingState state = IStaking(stakingInstance).getStakingState(serviceId);
-        if (state != IStaking.StakingState.Staked) {
-            revert ServiceNotStaked(stakingInstance, serviceId);
+        IServiceRegistry.ServiceState state;
+        (, multisig, , , , , state) = IServiceRegistry(serviceRegistry).mapServices(serviceId);
+        if (state != IServiceRegistry.ServiceState.Deployed) {
+            revert WrongServiceState(uint256(state), serviceId);
         }
     }
 
     /// @dev Checks for mech validity.
     /// @dev mech Agent mech contract address.
-    /// @param mechStakingInstance Agent mech staking instance address.
     /// @param mechServiceId Agent mech service Id.
-    /// @return multisig Service multisig address.
+    /// @return multisig
     function checkMech(
```

Comment:
probably can further simplify this method, but for now its ok

Reply:
This naturally resolves itself